Flaky tests in CI

### Product versie / Product version

latest

### Omschrijf het probleem / Describe the bug

Some tests in both the CMIS and non CMIS testsuite are flaky:
cmis:
https://github.com/open-zaak/open-zaak/actions/runs/13457130202/job/37605680162?pr=1910

```
======================================================================
ERROR: test_update_informatieobject_fails (openzaak.components.zaken.tests.test_validation_cmis.ZaakInformatieObjectValidationCMISTests.test_update_informatieobject_fails)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/zaken/tests/test_validation_cmis.py", line 87, in test_update_informatieobject_fails
    self.assertEqual(validation_error["code"], IsImmutableValidator.code)
                     ~~~~~~~~~~~~~~~~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

======================================================================
ERROR: test_partial_update_zaak_and_informatieobject_fails (openzaak.components.zaken.tests.test_zaakinformatieobjecten_cmis.ZaakInformatieObjectCMISAPITests.test_partial_update_zaak_and_informatieobject_fails) (field='informatieobject')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/zaken/tests/test_zaakinformatieobjecten_cmis.py", line 319, in test_partial_update_zaak_and_informatieobject_fails
    self.assertEqual(error["code"], IsImmutableValidator.code)
                     ~~~~~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

======================================================================
ERROR: test_update_zaak_and_informatieobject_fails (openzaak.components.zaken.tests.test_zaakinformatieobjecten_cmis.ZaakInformatieObjectCMISAPITests.test_update_zaak_and_informatieobject_fails) (field='informatieobject')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/zaken/tests/test_zaakinformatieobjecten_cmis.py", line 292, in test_update_zaak_and_informatieobject_fails
    self.assertEqual(error["code"], IsImmutableValidator.code)
                     ~~~~~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

======================================================================
FAIL: test_create_external_informatieobjecttype_fail_invalid_schema (openzaak.components.documenten.tests.test_enkelvoudiginformatieobject_cmis.InformatieobjectCreateExternalURLsTests.test_create_external_informatieobjecttype_fail_invalid_schema)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/documenten/tests/test_enkelvoudiginformatieobject_cmis.py", line 1024, in test_create_external_informatieobjecttype_fail_invalid_schema
    self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
AssertionError: 201 != 400

======================================================================
FAIL: test_create_external_informatieobjecttype_fail_non_pulish (openzaak.components.documenten.tests.test_enkelvoudiginformatieobject_cmis.InformatieobjectCreateExternalURLsTests.test_create_external_informatieobjecttype_fail_non_pulish)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/open-zaak/open-zaak/src/openzaak/components/documenten/tests/test_enkelvoudiginformatieobject_cmis.py", line 1070, in test_create_external_informatieobjecttype_fail_non_pulish
    self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
AssertionError: 201 != 400
```

regular:
https://github.com/open-zaak/open-zaak/actions/runs/13457735824/job/37605379936

-     openzaak.components.documenten.tests.test_auth.GebruiksrechtenReadTests.test_create_gebruiksrechten_limited_to_authorized_io
 


### Stappen om te reproduceren / Steps to reproduce

_No response_

### Verwacht gedrag / Expected behavior

_No response_

--- a/src/openzaak/components/zaken/tests/test_zaakobject_zaakobjecttype.py
+++ b/src/openzaak/components/zaken/tests/test_zaakobject_zaakobjecttype.py
@@ -127,11 +127,13 @@ class ZaakObjectZaakobjecttypeTestCase(JWTAuthMixin, APITestCase):
 
     def test_patch_zaakobject_change_zaakobjecttype_fail(self):
         zaakobjecttype = ZaakObjectTypeFactory.create()
+        other_zaakobjecttype = ZaakObjectTypeFactory.create()
         zaak = ZaakFactory.create(zaaktype=zaakobjecttype.zaaktype)
         zaakobject = ZaakObjectFactory.create(
             zaak=zaak,
             object=OBJECT,
             object_type=ZaakobjectTypes.adres,
+            zaakobjecttype=other_zaakobjecttype,
         )
         url = reverse(zaakobject)
 

--- a/src/openzaak/config/tests/test_feature_flags.py
+++ b/src/openzaak/config/tests/test_feature_flags.py
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: EUPL-1.2
 # Copyright (C) 2020 Dimpact
 from base64 import b64encode
+from unittest.mock import patch
 
 from rest_framework import status
 from rest_framework.test import APITestCase
@@ -21,6 +22,10 @@ from openzaak.tests.utils import JWTAuthMixin
 from ..models import FeatureFlags
 
 
+@patch(
+    "openzaak.utils.validators.FeatureFlags.get_solo",
+    return_value=FeatureFlags(allow_unpublished_typen=True),
+)
 class ConceptFeatureFlagTests(JWTAuthMixin, APITestCase):
     """
     Test the feature flags that bypass FOO.concept validation.
@@ -28,15 +33,7 @@ class ConceptFeatureFlagTests(JWTAuthMixin, APITestCase):
 
     heeft_alle_autorisaties = True
 
-    @classmethod
-    def setUpTestData(cls):
-        super().setUpTestData()
-
-        feature_flags = FeatureFlags.get_solo()
-        feature_flags.allow_unpublished_typen = True
-        feature_flags.save()
-
-    def test_zaak_create(self):
+    def test_zaak_create(self, mock_get_solo):
         """
         Assert that it's possible to create a zaak with an unpublished zaaktype when
         the feature flag is set.
@@ -56,7 +53,7 @@ class ConceptFeatureFlagTests(JWTAuthMixin, APITestCase):
 
         self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
 
-    def test_informatieobject_create(self):
+    def test_informatieobject_create(self, mock_get_solo):
         eio_url = reverse(EnkelvoudigInformatieObject)
         informatieobjecttype = InformatieObjectTypeFactory.create(concept=True)
         informatieobjecttype_url = reverse(informatieobjecttype)
@@ -81,7 +78,7 @@ class ConceptFeatureFlagTests(JWTAuthMixin, APITestCase):
         # Test response
         self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
 
-    def test_besluit_create(self):
+    def test_besluit_create(self, mock_get_solo):
         besluit_url = reverse(Besluit)
         besluittype = BesluitTypeFactory.create(concept=True)
         besluittype_url = reverse(besluittype)

